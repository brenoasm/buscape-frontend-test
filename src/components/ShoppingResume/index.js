import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import { colors } from "../../theme";

const propTypes = {};

const defaultProps = {};

const StyledShoppingResume = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  padding: 20px;

  span {
    font-weight: 700;
  }

  > div {
    align-self: stretch;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    padding-top: 8px;
    border-top: solid 1px ${colors.black};
  }
`;

const ShoppingResume = ({ subtotal }) => {
  return (
    <StyledShoppingResume>
      <span>subtotal</span>
      <div>
        <span>
          {subtotal.installments}x de R$ {subtotal.installmentValue}
        </span>
        <span>ou R$ {subtotal.value} Ã  vista</span>
      </div>
    </StyledShoppingResume>
  );
};

ShoppingResume.propTypes = propTypes;
ShoppingResume.defaultProps = defaultProps;

export default ShoppingResume;
