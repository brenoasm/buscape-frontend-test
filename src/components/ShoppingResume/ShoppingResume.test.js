import React from "react";
import { shallow } from "enzyme";

import ShoppingResume from "./";

describe("ShoppingResume", () => {
  const subtotal = {
    installments: 5,
    installmentValue: 500,
    value: 2500
  }

  it("matches the snapshot", () => {
    const wrapper = shallow(<ShoppingResume price={subtotal} />);

    expect(wrapper).toMatchSnapshot();
  });
});
