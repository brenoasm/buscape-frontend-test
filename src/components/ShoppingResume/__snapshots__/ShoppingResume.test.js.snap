// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingResume matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShoppingResume
    price={
      Object {
        "installmentValue": 500,
        "installments": 5,
        "value": 2500,
      }
    }
    subtotal={
      Object {
        "installmentValue": 0,
        "installments": 0,
        "value": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <span>
          subtotal
        </span>,
        <CartSubtotal
          price={
            Object {
              "installmentValue": 0,
              "installments": 0,
              "value": 0,
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "subtotal",
        },
        "ref": null,
        "rendered": "subtotal",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "price": Object {
            "installmentValue": 0,
            "installments": 0,
            "value": 0,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  padding: 20px;

  span {
    font-weight: 700;
  }

  > div {
    align-self: stretch;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    padding-top: 8px;
    border-top: solid 1px ",
          "#000",
          ";
  }
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <span>
            subtotal
          </span>,
          <CartSubtotal
            price={
              Object {
                "installmentValue": 0,
                "installments": 0,
                "value": 0,
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "subtotal",
          },
          "ref": null,
          "rendered": "subtotal",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "price": Object {
              "installmentValue": 0,
              "installments": 0,
              "value": 0,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  padding: 20px;

  span {
    font-weight: 700;
  }

  > div {
    align-self: stretch;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    padding-top: 8px;
    border-top: solid 1px ",
            "#000",
            ";
  }
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
